lib/cmd/fh3/appforms/environments/submissions/exportcsv.js

```diff
@@ -2,11 +2,92 @@
 var fhreq = require('../../../../../utils/request.js');
 var fs = require('fs');
 var util = require('util');
+var async = require('async');
+var _ = require('underscore');
+var log = require('../../../../../utils/log');
+
+
+/**
+ *
+ * Trying to export the submissions using asnyc Submission CSV export APIs.
+ *
+ * @param params
+ * @param cb
+ */
+function tryExportAsync(params, cb) {
+  var self = this;
+  var url = self.urlAsyncStart(params);
+
+  fhreq.POST(undefined, url, _.pick(params, 'fieldHeader', 'formId', 'projectId'), function(err, asyncStatus) {
+    log.silly("export status", err);
+    return cb(err, asyncStatus);
+  });
+}
+
+
+/**
+ *
+ * Calling the old synchronous submission csv export endpoint.
+ *
+ * @param params
+ * @param cb
+ */
+function doExportSync(params, cb) {
+  params.url = this.url(params);
+  params.method = this.method;
+
+  fhreq.downloadFile(params, cb);
+}
+
+
+/**
+ *
+ * Utility function to poll for the export status until it has completed or errored out.
+ *
+ * @param params
+ * @param cb
+ */
+function pollForExportStatus(params, cb) {
+  var self = this;
+  var url = self.urlAsyncStatus(params);
+  var prevMessage;
+
+  var csvExportPollInterval = setInterval(function() {
+    fhreq.GET(undefined, url, function(err, asyncStatus) {
+      asyncStatus = asyncStatus || {};
+      if(err) {
+        clearInterval(csvExportPollInterval);
+        return cb(err);
+      }
+
+      if(asyncStatus.status === 'inprogress') {
+        //Just updating the user on the status of the export
+
+        //Only log out a progress message if it has changed. Otherwise it is too verbose.
+        if(asyncStatus.message !== prevMessage) {
+          prevMessage = asyncStatus.message;
+          console.log(asyncStatus.message);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this logging statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)